### DISCUSSION TITLE
Some problems during URDF import process

### DISCUSSION THREAD

**Discussion by liuyidesktop** [stack-of-tasks/pinocchio#2179] [General]
I successfully imported the URDF file using the pinocchio library in c++ and was able to complete the relevant calculations. The code is as follows. 
    pinocchio::Model model;
    std::string urdf_filename = "/home/liuyi/Leg_robot/URDF_generate/test.urdf";
    pinocchio::JointModelFreeFlyer root_joint; 
    pinocchio::urdf::buildModel(urdf_filename, root_joint,model);
    pinocchio::Data data(model);
   But when I transplanted it to python, I got the following error after importing URDF，I use the command 
   conda install pinocchio -c conda-forge   to install Pinocchio , the version is 2.7.0

   model=pinocchio.buildModelsFromUrdf("/home/liuyi/文档/proj_robot/src/macro_arm6/urdf/macro_arm6.urdf")
   data= model.createData()
   
   output:
  AttributeError: 'tuple' object has no attribute 'createData'

**Comment by ManifoldFR** [comment]
Hi,

Yeah this function actually returns a tuple (it says buildModel**s** plural and returns the Model and at least a GeometryModel object).

The right version to call in Python would be buildModelFromUrdf (no extra **s** here)

**Reply by liuyidesktop** [reply]
Thank you for your reply, my problem is solved
